File: src/main/java/com/grizzly/baseViews/BaseFrameLayout.java
package com.grizzly.baseViews;

import android.annotation.TargetApi;
import android.app.Activity;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.widget.FrameLayout;

/**
 * Created by FcoPardo on 3/6/16.
 *
 * Base class for creating custom views using a FrameLayout.
 */
public abstract class BaseFrameLayout extends FrameLayout {

    protected int layout = 0;
    protected int parentType = BaseView.PARENT_UNKNOW;

    public BaseFrameLayout(Context context) {
        super(context);
        inflateBaseLayout();
    }

    public BaseFrameLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        inflateBaseLayout();
    }

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    public BaseFrameLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        inflateBaseLayout();
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public BaseFrameLayout(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        inflateBaseLayout();
    }

    protected void inflateBaseLayout(){
        setContainer();
        if(layout>0)BaseView.inflateLayout(layout, getContext(), this);
        inflateComponents();
    }

    protected abstract void inflateComponents();

    protected abstract void setContainer();

    protected Activity getActivity(){
        return BaseView.getActivity(this);
    }

    public int getParentType() {
        return parentType;
    }

    public <T extends BaseFrameLayout> T setParentType(@BaseView.ParentType int parentType) {
        this.parentType = parentType;
        return (T) this;
    }
}

File: src/main/java/com/grizzly/baseViews/BaseFrameLayout.kt
package com.grizzly.baseViews

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout

/**
 * Created by FcoPardo on 4/29/18.
 */
open abstract class BaseFrameLayout : FrameLayout, BaseView.OnVariableView {

    override var layout = 0
    override var aParentType = BaseView.PARENT_UNKNOW

    constructor(context : Context) : super(context){
        inflateBaseLayout()
    }
    constructor(context : Context, attrs : AttributeSet) : super(context, attrs) {
        inflateBaseLayout()
    }
    constructor(context : Context, attrs : AttributeSet, styleAttr : Int) : super(context, attrs, styleAttr){
        inflateBaseLayout()
    }
    constructor(context : Context, attrs : AttributeSet, styleAttr : Int, defStyleAttr : Int) : super(context, attrs, styleAttr, defStyleAttr){
        inflateBaseLayout()
    }

}